lib/sentry_logging.rb

```diff
@@ -1,6 +1,74 @@
 # frozen_string_literal: true
 
 module SentryLogging
+  SKIP_SENTRY_EXCEPTION_TYPES = [
+    Common::Exceptions::Unauthorized,
+    Common::Exceptions::RoutingError,
+    Common::Exceptions::Forbidden,
+    Breakers::OutageException
+  ].freeze
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def log_error(error)
+    if skip_sentry_exception_types.include?(error.class)
+      Rails.logger.error error.message.to_s, backtrace: error.backtrace
+    else
+      extra = error.respond_to?(:errors) ? { errors: error.errors.map(&:to_hash) } : {}
+      if error.is_a?(Common::Exceptions::BackendServiceException)
+        # Add additional user specific context to the logs
+        if @current_user.present?
```

Comment:
Should this be a responsibility of the `Common::Exceptions::BackendServiceException`?

Reply:
`Common::Exceptions::BackendServiceException` is raised in a number of places, and is a parent of class of a few of other error classes , many/most of which don't have access to the `@current_user` object.
That all raises the following question for me:  Should this code exist here, or in a library explicitly intended for use in `controller`s (or just refactored in the `ApplicationController`?  This library is used/included in controllers, AR models, Redis models, service classes, Sidekiq workers, etc, and most/all of this explicitly deals with `controller` concerns (request objects, current_user objects, etc)

Reply:
Yes, I think so - the logic here is one of the most confusing aspects of errors across the platform and I think extracting the context building to the caller as Sentry doesn't (or shouldn't) care/know about any context specifics it just logs hashes of data

Reply:
I think if we were to move this responsibility into the `::BackendServiceException` class it would be obfuscated, we may want more explicitness around the adding of user context info to Sentry